merge/HILD_merge.R

```diff
@@ -0,0 +1,654 @@
+		rl_ssn1 <- rlink(df = df_sub,
+			block = "ssn1",
+			string = s,
+			phonetic = p
+			))
+	gc()
+
+### dob1_ym RL
+	system.time(
+		rl_dob1_ym <- rlink(df = df_sub,
+			block = c("dob_y","dob1_m"),
+			string = s,
+			phonetic = p,
+			name_wt = "ym_wt"
+			))
+	gc()
+
+### fname RL
+	system.time(
+		rl_fname <- rlink(df = df_sub,
+			block = c("fname"),
+			string = s,
+			phonetic = p,
+			threshold = .1,
+			name_wt = "fname_wt"
+			))
+	gc()
+
+### lname RL
+	system.time(
+		rl_lname <- rlink(df = df_sub,
+			block = c("lname"),
+			string = s,
+			phonetic = p,
+			threshold = .1,
+			name_wt = "lname_wt"
+			))
+	gc()
+
+# ==========================================================================
+# Create link df
+# ==========================================================================
+
+### Join links
+	link <- rl_ssn1 %>%
+			select(ssn.1:ssn_dq.1,
+				   pid0.1,
+				   pid0.2,
+				   pid1.1,
+				   pid1.2,
+				   pid2.1,
+				   pid2.2,
+				   ends_with("_wt"))
+
+	link <- full_join(link,
+					  rl_dob1_ym %>%
+			select(pid2.1,
+				   pid2.2,
+				   ends_with("_wt")))
+
+	link <- full_join(link,
+					  rl_fname %>%
+			select(pid2.1,
+				   pid2.2,
+				   ends_with("_wt")))
+
+	link <- full_join(link,
+					  rl_lname %>%
+			select(pid2.1,
+				   pid2.2,
+				   ends_with("_wt")))
+
+### Change NA to 0
+	link <- link %>%
+			mutate_at(vars(ends_with("wt")),
+					 	   funs(ifelse(is.na(.)==T,
+					  	   0.0,
+					  	   .))
+					  )
+	glimpse(link)
+
+# ==========================================================================
+# Create product weights
+# ==========================================================================
+
+### Weighting function
+	wtp <- function(p,weights){
+			1-prod((1-p)^weights)^(1/sum(weights))
+		}
+
+### Different weighting scenerios
+	w1 <- link %>%
+		mutate(wtp111 = apply( # apply function across rows
+						select(., ends_with("wt")), # select columns to apply to
+						1, # w/in each row (2 would do it w/in each col)
+						wtp, # function to apply
+						weights=c(1,1,1,1)), # weights argument
+			   wtp1.8.8 = apply( # apply function across rows
+						select(., ends_with("wt")), # select columns to apply to
+						1, # w/in each row (2 would do it w/in each col)
+						wtp, # function to apply
+						weights=c(1,.8,.8,.8)), # weights argument
+			   wtp1.5.5 = apply( # apply function across rows
+						select(., ends_with("wt")), # select columns to apply to
+						1, # w/in each row (2 would do it w/in each col)
+						wtp, # function to apply
+						weights=c(1,.5,.5,.5)) # weights argument
+						)
+
+	head(w1) # weights all at 1
+	hist(w1$wtp111, breaks = 40)
+	hist(w1$wtp1.8.8, breaks = 40)
+	hist(w1$wtp1.5.5, breaks = 40)
+
+# ==========================================================================
+# Save file
+# ==========================================================================
+
+	write.csv(w1, "data/Housing/links/WeightedLinks_20180402.csv")
```

Comment:
Please use absolute paths. 

We'll also probably need to figure out a better way to version files, rather than giving them a name with a date. Either way, can you write code to put *today's* date (where today is whenever you are running the code) here, instead of hard coding some particular date?